tsl/src/compression/create.c

```diff
@@ -888,8 +888,15 @@ disable_compression(Hypertable *ht, WithClauseResult *with_clause_options)
 		return false;
 	/* compression is enabled. can we turn it off? */
 	check_modify_compression_options(ht, with_clause_options);
-	drop_existing_compression_table(ht);
-	ts_hypertable_unset_compressed_id(ht);
+
+	/* distributed hypertables do not have compression table on the access node */
+	if (TS_HYPERTABLE_HAS_COMPRESSION_TABLE(ht))
+		drop_existing_compression_table(ht);
```

Comment:
Given that `ts_hypertable_unset_compressed()` is called in the `else` case, I would have expected it is called also when you have a compression table. In fact, to unset the compression state seems to be independent of this if-clause. 

For clarity, I would move `ts_hypertable_unset_compressed()` out of this if clause and out of `drop_existing_compression_table` to the end of this function. Then it is set at the same level for both cases and it is clear it is always unset.

Reply:
I did consider the option of making unset_compressed common to both paths. drop_existing_compression_table is also called from other places in the code.  So it makes sense to encapsulate all the things that need to happen during a drop in a single function.